radix-engine/src/vm/wasm_runtime/scrypto_runtime.rs

```diff
@@ -794,4 +794,25 @@ impl<'y, Y: SystemApi<RuntimeError>> WasmRuntime for ScryptoRuntime<'y, Y> {
 
         self.allocate_buffer(key.to_vec())
     }
+
+    /// This method is only available to packages uploaded after "Cuttlefish"
+    /// protocol update due to checks in [`ScryptoV1WasmValidator::validate`].
+    #[trace_resources]
+    fn crypto_utils_secp256k1_ecdsa_verify_and_key_recover_uncompressed(
+        &mut self,
+        message: Vec<u8>,
+        signature: Vec<u8>,
+    ) -> Result<Buffer, InvokeError<WasmRuntimeError>> {
+        let hash = Hash::try_from(message.as_slice()).map_err(WasmRuntimeError::InvalidHash)?;
+        let signature = Secp256k1Signature::try_from(signature.as_ref())
+            .map_err(WasmRuntimeError::InvalidSecp256k1Signature)?;
+
+        self.api
+            .consume_cost_units(ClientCostingEntry::Secp256k1EcdsaKeyRecover)?;
```

Comment:
Is this `Secp256k1EcdsaKeyRecover` variant used here deliberately?
I guess the costs for uncompressed variant are similar, so this probably makes sense.

Reply:
I saw this too. The uncompressed version should be cheaper, but didn't bother changing it.